samza-core/src/main/java/org/apache/samza/zk/ZkLeaderElector.java

```diff
@@ -49,28 +49,34 @@
   private final String processorIdStr;
   private final ZkKeyBuilder keyBuilder;
   private final String hostName;
-  private final ScheduleAfterDebounceTime debounceTimer;
 
   private AtomicBoolean isLeader = new AtomicBoolean(false);
-  private IZkDataListener previousProcessorChangeListener;
+  private final IZkDataListener previousProcessorChangeListener;
+  private LeaderElectorListener leaderElectorListener = null;
   private String currentSubscription = null;
   private final Random random = new Random();
 
-  @VisibleForTesting
-  public void setPreviousProcessorChangeListener(IZkDataListener previousProcessorChangeListener) {
-    this.previousProcessorChangeListener = previousProcessorChangeListener;
-  }
-
-  public ZkLeaderElector(String processorIdStr, ZkUtils zkUtils,  ScheduleAfterDebounceTime debounceTimer) {
+  public ZkLeaderElector(String processorIdStr, ZkUtils zkUtils) {
     this.processorIdStr = processorIdStr;
     this.zkUtils = zkUtils;
     this.keyBuilder = zkUtils.getKeyBuilder();
     this.hostName = getHostName();
-    this.debounceTimer = (debounceTimer != null) ? debounceTimer : new ScheduleAfterDebounceTime();
+    this.previousProcessorChangeListener = new PreviousProcessorChangeListener();
 
     zkUtils.makeSurePersistentPathsExists(new String[]{keyBuilder.getProcessorsPath()});
   }
 
+  @VisibleForTesting
```

Comment:
yeah, I wish we didn't have to do this. Don't have any suggestions at the moment now, but if you do... :)